src/raft/progress.rs

```diff
@@ -41,6 +41,43 @@ impl Default for ProgressState {
     }
 }
 
+/// `ProgressSet` contains several `Progress`es,
+/// which could be `Leader`, `Follower` and `Learner`.
+#[derive(Default, Clone)]
+pub struct ProgressSet {
+    pub voters: FlatMap<u64, Progress>,
+    pub learners: FlatMap<u64, Progress>,
+}
+
+impl ProgressSet {
```

Comment:
Could you provide a method like `fn for_each<F: FnMut(u64, &mut Progress, bool)>(&mut self, func: F)`? 
So that we do not need to write something like 
```
voter.iter().for_each(do_something); 
leader.iter().for_each(do_something);
```

Reply:
I thinks we don't need that. If we want to apply same closure on `voters` and `learners`, we can call `voters.iter().chain(&learners).for_each(...)` directly.